src/components/fields/number-field/number-field.spec.js

```diff
@@ -1,172 +1,210 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import PropTypes from 'prop-types';
+import omit from 'lodash.omit';
+import { render, fireEvent } from '../../../test-utils';
 import NumberField from './number-field';
-import FieldLabel from '../../field-label';
-import NumberInput from '../../inputs/number-input';
-import FieldErrors from '../../field-errors';
-import { AddBoldIcon } from '../../icons';
-
-const createTestProps = customProps => ({
-  title: 'Username',
-  value: '',
-  onChange: () => jest.fn(),
-  ...customProps,
-});
-
-describe('rendering', () => {
-  describe('data attributes', () => {
-    let textInput;
-    beforeEach(() => {
-      const props = createTestProps({
-        'data-foo': 'bar',
-        'data-test': 'baz',
-      });
-      const wrapper = shallow(<NumberField {...props} />);
-      textInput = wrapper.find(NumberInput);
-    });
-    it('should forward the attributes to the NumberInput', () => {
-      expect(textInput).toHaveProp('data-foo', 'bar');
-      expect(textInput).toHaveProp('data-test', 'baz');
-    });
+
+// This component is used to enable easy testing.
+// It overwrites the onChange function and places a label for the
+// input component. It also ensures an id so that the label can associate
+// the input. This allows tests to use getByLabelText.
+// It also makes sure the event's value passed to onChange flows back to the
+// component so that we can test it under real conditions.
+// As a convenience, we enable accessing a mocked onChange function.
+class Story extends React.Component {
+  static displayName = 'Story';
+  static propTypes = {
+    onEvent: PropTypes.func.isRequired,
+    value: PropTypes.string,
+    id: PropTypes.string,
+  };
+  static defaultProps = {
+    id: 'text-field',
+  };
+  state = {
+    value: this.props.value || '',
+  };
+  handleChange = event => {
+    this.props.onEvent(event);
+    this.setState({ value: event.target.value });
+  };
+  render() {
+    return (
+      <div>
+        <label htmlFor={this.props.id}>NumberField</label>
+        <NumberField
+          id={this.props.id}
+          {...omit(this.props, 'onEvent')}
+          value={this.state.value}
+          onChange={this.handleChange}
+        />
+      </div>
+    );
+  }
+}
+
+const renderNumberField = (customProps, options) => {
+  const onChange = jest.fn();
+  const props = {
+    title: 'foo',
+    ...customProps,
+  };
+  return {
+    onChange,
+    ...render(<Story onEvent={onChange} {...props} />, options),
```

Comment:
Nitpick because I'm afraid this will get copied around: I would recommend to put the spread first and then provide the `onChange` to avoid accidental overwrites (in other cases):

```js
return {
  ...render(<Story onEvent={onChange} {...props} />, options),
  onChange
}
```